src/ray/raylet/worker_pool.cc

```diff
@@ -446,20 +447,25 @@ boost::optional<const rpc::JobConfig &> WorkerPool::GetJobConfig(
 }
 
 Status WorkerPool::RegisterWorker(const std::shared_ptr<WorkerInterface> &worker,
-                                  pid_t pid,
+                                  pid_t pid, pid_t worker_shim_pid,
                                   std::function<void(Status, int)> send_reply_callback) {
   RAY_CHECK(worker);
 
   auto &state = GetStateForLanguage(worker->GetLanguage());
-  auto process = Process::FromPid(pid);
-
-  if (state.starting_worker_processes.count(process) == 0) {
-    RAY_LOG(WARNING) << "Received a register request from an unknown worker "
-                     << process.GetId();
+  // When the "shim process" setup worker is not needed, the worker_shim_pid will be 0.
+  if (worker_shim_pid == 0) {
+    worker_shim_pid = pid;
```

Comment:
should it always be set?

Reply:
@iycheng  The `worker_shim_pid` will be set only if enable runtime env. https://github.com/alipay/ray/blob/refact_worker_pid_validate/python/ray/workers/setup_runtime_env.py#L39

Reply:
It can be set when creating the register message in RayletClient.

Reply:
I think `worker_shim_pid` should not be set if we don't use shim process

Reply:
I have moved this logic into RayletClient